app/models/ability.rb

```diff
@@ -0,0 +1,44 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    user ||= User.new # Guest user
+
+    if user.admin?
+      can :manage, :all # Admins can manage all posts
+    else
+      can :manage, Post, author_id: user.id # Users can manage their own posts
+      can :manage, Comment, user_id: user.id # Users can manage their own comments
+    end
+
+    return unless user.user?
+
+    can :read, Post
+    can :read, Comment
+  end
```

Comment:
- [ ] Please add authorization like this `A user can delete a post if it is theirs or if they have an admin role`
- [ ] Please add authorisation like this `A user can delete a comment if it is theirs or if they have an admin role`
- [ ] But in your case the admin is not allowed to the other's post so kindly fix it
- [ ] Check below :point_down:  the video for more details :slightly_smiling_face: 

_**Please play this video on your Chrome Brower because it is not played on another browser I don't know why so kindly play this video on Chrome Browser**_

[screen-capture (28).webm](https://github.com/danielochuba/duo-blog/assets/45226848/4be66469-de31-4031-b1f5-35dfaf285d66)